File: go/difference-of-squares/difference_of_squares.go
package diffsquares

// Returns the square of the sum of first n natural numbers
// Sum of n natural numbers is
// S=(1/2)(n)(n+1).
func SquareOfSum(n int) int {
	sum := n * (n + 1) / 2
	return sum * sum
}

func SumOfSquares(n int) int {
	sum := 0
	for i := 1; i <= n; i++ {
		sum += i * i
	}
	return sum
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: typescript/difference-of-squares/difference-of-squares.ts
export class Squares {
    private count: number;
    constructor(count: number) {
        this.count = count;
    }

    get sumOfSquares(): number {
        // [n(n + 1)(2n + 1)] / 6
        return (this.count * (this.count + 1) * (2 * this.count + 1)) / 6;
    }

    get squareOfSum(): number {
        // (n(n+1) / 2) ^ 2
        return ((this.count * (this.count + 1)) / 2) ** 2;
    }

    get difference(): number {
        return this.squareOfSum - this.sumOfSquares;
    }
}
